styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -570,6 +571,15 @@ private void setupMetrics(
           workflow.id(),
           () -> stateManager.getActiveStatesCount(workflow.id()));
 
+      final Optional<Workflow> existingWorkflow = cache.workflow(workflow.id());
+      if (existingWorkflow.isPresent()) {
+        // if we have ever registered/updated the workflow from API,
+        // we ignore any subsequent updates via schedule source
+        if (existingWorkflow.get().fromApi().isPresent() && !workflow.fromApi().isPresent()) {
```

Comment:
This line reads a bit hard if you're not familiar with the history of this field. What you want to say is, "was this call made through the new API?", but I would read this as "was this call made through an unknown API version?". Ideally we would want to easily compare API versions, both to clarify intent and for future-proofing. For example:

```java
if (isHigher(existingWorkflow.get().fromApi(), workflow.fromApi())) {
  return;
}
```